index.js

```diff
@@ -1,84 +1,170 @@
-        var interval = urls[name].interval;
-        createloop(name, url, interval);
+    for (const [name, params] of Object.entries(urls)) {
+        const url = params.url;
+        const interval = params.interval;
+        const dateBounds = {
+            minTimeOfDay: params.minTimeOfDay,
+            maxTimeOfDay: params.maxTimeOfDay,
+            daysOfTheWeek: params.daysOfTheWeek
+        }
+        const mentionsWhenDown = params.mentionsWhenDown;
+        createLoop(name, url, interval, dateBounds, mentionsWhenDown);
     }
 });
 
 /* Create a loop that's executed every x milliseconds */
-function createloop(name, url, interval) {
-    var requestLoop = setInterval(function () {
-        request({
+function createLoop(name, url, interval, dateBounds, mentionsWhenDown) {
+    setInterval(function () {
+        const requestDate = new Date();
+        if (!isDateWithinBounds(requestDate, dateBounds)) {
+            urlContainer[name] = undefined;
+            return;
+        }
+
+        request.get({
             url: url,
-            method: "GET",
-            timeout: 10000,
-            followRedirect: true,
-            maxRedirects: 10
-        }, function (error, response, body) {
-            if (!error && response.statusCode == 200) {
-                if (typeof notReachable[name] != "undefined") {
-                  if (notReachable[name] == "notagain") {
-                    delete notReachable[name];
-                    var message = name + " " + emoji.get('beers') + " " + getFormattedDate();
-                    send(message);
-                    console.log(message);
-                  }
-                }
+            timeout: 2000,
+            followRedirect: true
+        }, (error, response, body) => {
+            const responseDate = new Date();
+
+            if (error !== null) {
+                handleLocalError(name, responseDate);
+            } else if (response.statusCode == 200) {
+                handleSuccessStatus(name, url, responseDate);
             } else {
-                if (typeof notReachable[name] == "undefined") { // first time
-                    notReachable[name] = "true";
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log("first time :" + message);
-
-                } else if (notReachable[name] == "true") { // secondtime
-                    notReachable[name] = "notagain"; // we don't want second notification
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log(message);
-                    send(message);
-                }
+                handleDownStatus(name, url, responseDate, mentionsWhenDown);
             }
         });
     }, interval);
 }
 
+function handleSuccessStatus(urlName, urlValue, responseDate) {
+    const message = `:globe_with_meridians: ${urlName} (${urlValue}) is up and running!`;
```

Comment:
this is only used in a branch, move it there.

also, not for the first PR, but it may be worth to contribute back to the public project a i18n version, where we could use identifiers and people can easily change the strings by changing the translation file (even if in english, suppose some company is not happy with some of the emojis)